openmdao/components/tests/test_submodel_comp.py

```diff
@@ -440,3 +440,22 @@ def test_add_io_meta_override(self):
         self.assertEqual(p.get_val('new_r'), 20)
         self.assertEqual(p.get_val('new_theta'), 0.5)
         self.assertEqual(p.get_val('new_x'), 100)
+
+    def test_subprob_solver_print(self):
+        p = om.Problem()
+        subprob = om.Problem()
+        subprob.model.add_subsystem('comp', om.ExecComp('x = r*cos(theta)'), promotes=['*'])
+        submodel = om.SubmodelComp(problem=subprob)
+
+        submodel.add_input('r', name='new_r', val=20)
+        submodel.add_input('theta', name='new_theta', val=0.5)
+        submodel.add_output('x', name='new_x', val=100)
+
+        submodel.set_subprob_solver_print(level=3, depth=20, type_='NL')
```

Comment:
We already have an instance of subprob to work with here, just use `subprob.set_solver_print(level=3, depth=20, type_='NL')` in this test.